Products/ZenModel/MaintenanceWindow.py

```diff
@@ -645,6 +647,31 @@ def _setProdState(devices_batch):
                 else:
                     device.setProdState(minProdState, maintWindowChange=True)
 
+        def retrySingleDevices(devices_batch):
+            log.warn("Retrying devices individually")
+            for dev in devices_batch:
+                try:
+                    processFunc([dev])
+                except:
+                    log.exception("The production stage change for %s raised the exception." % str(dev))
+                    unchangedDevices.append(dev)
+
+        def processBatchOfDevices(devices, batchSize):
+            for i in xrange(0, len(devices), batchSize):
+                log.info('MW %s processing batch #%s', self.displayName(), i / batchSize + 1)
+                dev_chunk = devices[i:i + batchSize]
+                try:
+                    processFunc(dev_chunk)
+                except (ConflictError, POSKeyError, ReadConflictError) as e:
+                    log.warn("While processing batch of %d devices exception was raised. %s" % (len(dev_chunk), e))
+                    retrySingleDevices(dev_chunk)
+                except:
```

Comment:
Please write this as `except Exception:`
Bare exception handlers will catch exceptions we do not want caught.

Reply:
fixed